WorkloadManagementSystem/JobWrapper/JobWrapper.py

```diff
@@ -1125,35 +1127,31 @@ def finalize(self):
     requests = self.__getRequestFiles()
     outputDataRequest = self.failoverTransfer.getRequest()
 
-    # try to send the failover request
-    res = self.sendFailoverRequest()
-    if not res['OK']:  # This means that the request could not be set (this should "almost never" happen)
-      self.__report(JobStatus.FAILED, minorStatus='Failed sending requests')
-      self.__cleanUp()
-      return 1
-
-    requestFlag = res['Value'] or len(requests) > 0 or not outputDataRequest.isEmpty()
-    if requestFlag:
-      self.log.info("Job finished with pending requests")
-      self.__report(minorStatus='Pending Requests')
-    else:
-      self.log.info("Job finished with no pending requests")
+    requestFlag = len(requests) > 0 or not outputDataRequest.isEmpty()
+    self.log.info("Job finished with%s pending requests" % ('' if requestFlag else ' no'))
 
     if self.failedFlag:
       self.log.info("Job finished with errors")
-      self.__report(JobStatus.FAILED)
-      self.__cleanUp()
-      return 1
-
-    self.log.info("Job finished successfully")
-    if requestFlag:
-      self.__report(JobStatus.COMPLETED)
+      self.__report(status=JobStatus.FAILED, minorStatus=JobMinorStatus.PENDING_REQUESTS if requestFlag else '')
     else:
-      self.__report(minorStatus='Execution Complete')
-      self.__report(JobStatus.DONE)
+      self.log.info("Job finished successfully")
+      if requestFlag:
+        self.__report(status=JobStatus.COMPLETED, minorStatus=JobMinorStatus.PENDING_REQUESTS if requestFlag else '')
+      else:
+        self.__report(status=JobStatus.DONE, minorStatus=JobMinorStatus.EXEC_COMPLETE)
+
+    # try to send the failover request if any, and send accounting
+    res = self.sendFailoverRequest()
+    if not res['OK']:  # This means that the request could not be set (this should "almost never" happen)
+      self.__report(status=JobStatus.FAILED, minorStatus=JobMinorStatus.FAILED_SENDING_REQUESTS)
+      self.failedFlag = True
+    elif res['Value'] and not requestFlag:
+      # A request was created while there were none before, change final status
+      self.__report(status=JobStatus.COMPLETED if self.wmsMajorStatus == JobStatus.DONE else '',
```

Comment:
This will result in the job status Done followed by Completed in the job logging which is not logically correct and is confusing

Reply:
This is an extremely rare case and as only the accounting would be missing, we can leave the job in the `Done` status...

Reply:
Change committed, thanks!

Reply:
Well that would become problematic if we introduce a state machine in which we declare that "Done" is a final (and immovable) state, as it should.

Reply:
Right, fixed as suggested by Andrei... We would have `Done;Pending Requests` for this (hopefully) very rare case in which anyway only the accounting was not successfully done...